test/e2e/setup/setup.go

```diff
@@ -238,6 +240,22 @@ func createOrUpdate(obj client.Object) error {
 	return nil
 }
 
+// withCPURequest assumes that the underlying type is an appsv1.Deployment.
```

Comment:
Can we maybe log something in case this is not a deployment? Or maybe even fail loudly so that is clear that this function has been called on the wrong type?

Reply:
Good idea, I added the error management to the options functions